app/scripts/directives/overview/virtualMachineRow.js

```diff
@@ -23,6 +23,39 @@
     templateUrl: 'views/overview/_virtual-machine-row.html'
   });
 
+  function fileDownload(data, fileName, mimeType) {
+    fileName = fileName || "vm.rdp";
+    mimeType = mimeType || "application/rdp";
+
+    var a = document.createElement('a');
+    if (navigator.msSaveBlob) { // IE10
+      return navigator.msSaveBlob(new Blob([data], { type: mimeType }), fileName);
+    } else if ('download' in a) { // html5 A[download]
+      a.href = 'data:' + mimeType + ',' + encodeURIComponent(data);
+      a.setAttribute('download', fileName);
+      document.body.appendChild(a);
+      a.click();
+      document.body.removeChild(a);
+      return true;
+    } else { // do iframe dataURL download (old ch+FF):
+      var f = document.createElement('iframe');
+      document.body.appendChild(f);
+      f.src = 'data:' + mimeType + ',' + encodeURIComponent(data);
+      setTimeout(function () {
+        document.body.removeChild(f);
+      }, 333);
+      return true;
+    }
+  }
+
+  function buildRdp(address) {
+    return '' +
+      '\nfull address:s:' + address +
+      '\ndesktopwidth:i:1024' +
+      '\ndesktopheight:i:768' +
+      '';
```

Comment:
This will be extended by next force-push to enforce the defaults (they vary between client versions)